src/odemis/acq/align/autofocus.py

```diff
@@ -114,6 +115,28 @@ def MeasureOpticalFocus(image):
     return cv2.Laplacian(image, cv2.CV_64F).var()
 
 
+def Measure1d(image):
+    """
+    Given an image of a 1 line ccd, measure the focus based on the inverse of the width of a gaussian fit of the data.
+    It is assumed that the signal is in focus when the width of the signal is smallest and therefore sigma is smallest.
+    image (model.DataArray): 1D image from 1 line ccd.
+    returns (float): The focus level of the image.
+    """
+    def gauss(x, *p):
+        a, b, c, d = p
+        y = a * numpy.exp(-numpy.power((x - b), 2.) / (2. * c ** 2.)) + d
+        return y
+    # squeeze to make sure the image array is 1d.
+    signal = numpy.squeeze(image)
+    x = numpy.arange(len(signal))
+    # give an initial estimate for the parameters of the gaussian fit: [max, mu, sigma, y-offset]
+    p_initial = [max(signal), numpy.argmax(signal), 10, min(signal)]
```

Comment:
BTW, use `signal.max()` and `signal.min()` instead of the standard Python functions. This avoids the conversion to Python values for each elements... and makes it 10x faster.